src/lib/server/cart.ts

```diff
@@ -142,6 +156,8 @@ export async function addToCartInDb(
 			params.customPrice.amount,
 			toCurrency(params.customPrice.currency, product.price.amount, product.price.currency)
 		);
+	} else if (variationPriceDelta > 0) {
+		params.customPrice = { amount: variationPriceDelta, currency: runtimeConfig.mainCurrency };
```

Comment:
`customPrice` is not a delta, so it should be in addition to the original product price